maven-core/src/main/java/org/apache/maven/execution/ProjectActivation.java

```diff
@@ -0,0 +1,173 @@
+package org.apache.maven.execution;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import static java.util.stream.Collectors.toSet;
+
+/**
+ * Container for storing the request from the user to activate or de-activate certain projects and optionally fail the
+ * build if those projects do not exist.
+ */
+public class ProjectActivation
+{
+    private final Map<String, ActivationSettings> activations = new HashMap<>();
+
+    /**
+     * Adds a project activation to the request.
+     * @param selector The selector of the project.
```

Comment:
You probably should describe at least at class level what a selector is . It took me a while that it is either a `module`, `:artifactId` or `groupId:artifactId`.

Reply:
I agree, it was not defined anywhere near this class (if at all). I've added it to each method where "selector" is mentioned. Do you think this is too much and rather only have it at class-level instead?

Reply:
Make it class level or maybe we have definition at maven-site?

Reply:
For now, we've moved it to class level. If we find a nice spot in maven-site we can still add it there.

Reply:
Accepted.